src/game/gui/text/text.c

```diff
@@ -93,6 +99,27 @@ void text_free(text **t) {
     }
 }
 
+text_document *text_document_create(void) {
+    text_document *d = omf_calloc(1, sizeof(text_document));
+    vector_create(&d->text_objects, sizeof(text));
```

Comment:
vector_create_with_size_cb with text_free for free() callback ? That would drop the foreach loop from text_document_free().

Reply:
text_free takes a **text, which `vector_create_with_size_cb` wants a single layer of pointer indirection

Reply:
Elsewhere I just handled that by adding an intermediate function for the free operation that takes care of the indirection.